docs/Tutorials/CCE.md

```diff
@@ -0,0 +1,122 @@
+    might be a little low for full precision. lmax 16, filter 14 should be
+    pretty good, and typically precision doesn't improve above lmax 24,
+    filter 22 (be sure to update the filter as you update lmax -- it should
+    generally be around 2 lower than the maximum l to reduce possible aliasing).
+  - If you want to just run through the end of the provided worldtube data,
+    you can just omit the `EndTime` option and the executable will figure it
+    out from the worldtube file.
+  - The `ScriOutputDensity` adds extra interpolation points to the output,
+    which is useful for finite-difference derivatives on the output data, but
+    otherwise it'll just unnecessarily inflate the output files, so if you
+    don't need the extra points, best just set it to 1.
+  - If you're extracting at 100M or less, best to reduce the `TargetStepSize`,
+    to around .5 at 100M and lower yet for nearer extraction.
+  - The `InitializeJ` in the example file uses `InverseCubic` which is a pretty
+    primitive scheme, but early tests indicate that it gives the best results
+    for most systems.
+    If initial data is a concern, you can also try replacing the `InverseCubic`
+    entry with:
+    ```
+    NoIncomingRadiation:
+      AngularCoordTolerance: 1.0e-13
+      MaxIterations: 500
+      RequireConvergence: true
+    ```
+    which are probably pretty good choices for those parameters,
+    and the `RequireConvergence: true` will cause the iterative solve in
+    this version to error out if it doesn't find a good frame.
+  .
+- An example of an appropriate submission command for slurm systems is:
+  ```
+  srun -n 1 -c 1 path/to/build/bin/CharacteristicExtract ++ppn 3 \
+ --input-file path/to/input.yaml
+  ```
+  CCE doesn't currently scale to more than 4 cores, so those slurm options are
+  best.
+- CCE will work faster if the input worldtube hdf5 file is chunked in small
+  numbers of complete rows.
+  This is relevant because by default, SpEC writes its worldtube files
+  chunked along full time-series columns, which is efficient for
+  compression, but not for reading in to SpECTRE -- in that case,
+  it is recommended to rechunk the input file before running CCE
+  for maximum performance. This can be done, for instance, using h5py:
+  ```py
+  with h5py.File(input_file,'r') as input_h5,\
+    h5py.File(output_file, 'w') as output_h5:
+    for dset in input_h5:
+        if("Version" in dset):
+            output_h5[dset] = input_h5[dset][()]
+            continue
+        number_of_columns = input_h5[dset][()].shape[1]
+        output_h5.create_dataset(dset, data=input_h5[dset],
+                                 maxshape=(None, number_of_columns),
+                                 chunks=(4, number_of_columns), dtype='d')
+        for attribute in input_h5[dset].attrs.keys():
+            output_h5[dset].attrs[attribute] = input_h5[dset].attrs[attribute]
+  ```
+- The output data will be written as spin-weighted spherical harmonic
```

Comment:
Please describe what physical quantities are available.   Please provide an `h5ls` command to view what is in the files.
Please provide a python example of plotting a mode of strain data

Reply:
Maybe a very basic analysis script too?
```python
import numpy as np
import h5py
import matplotlib.pyplot as plt

# The CCE worldtube radii to open
radii = [50, 167, 283, 400]

# Print all the available data
print(h5py.File("CharacteristicExtractVolume" + str(radii[0]) +
                                "0.h5", 'r').keys())

# Print legend for the Strain _or_ add a function that computes the 1d index for the l,m pair
file_legend = h5py.File("CharacteristicExtractVolume" + str(radii[0]) +
                                "0.h5", 'r').get("Strain.dat").attrs['Legend']

# Loop over all radii and print the Psi_0 (2,2) mode
for radius in radii:
    data = np.asarray(h5py.File("CharacteristicExtractVolume" + str(radius) +
                                "0.h5", 'r').get("Psi0.dat"))
    plt.plot(data[:,0], data[:,17], label="Radius: " + str(radius))

plt.legend()
plt.savefig("Psi0_22.pdf")
```